dynawo/sources/Modeler/DataInterface/PowSyblIIDM/DYNInjectorInterfaceIIDM.h

```diff
@@ -179,7 +179,7 @@ class InjectorInterfaceIIDM {
   const powsybl::iidm::Injection& injectorIIDM_;                  ///< reference to the iidm injector instance
   boost::shared_ptr<BusInterface> busInterface_;                  ///< busInterface of the bus where the injector is connected
   boost::weak_ptr<VoltageLevelInterface> voltageLevelInterface_;  ///< voltageLevel interface where the injector is connected
-  std::string injectorId_;                                        ///< injector's id
+  const std::string& injectorId_;                                 ///< injector's id
```

Comment:
Please update the signature of the methode getID() to override the getID from DYNLoadInterface.h

Reply:
I cannot get rid of GetID() in Load, since it is a pure virtual method; I think it has to be instantiated within derived class. And I cannot get rid of GetID() in the Injector since it is used in trace log. Keeping a const ref makes the objects is not duplicated. I think we should let it this way. In both cases (Load and Injector) the ID is the one within powsybl object. One object, 2 methods for accessing the same object. I'm not sure having undertood what you ask. If it's confusing I can change getID() into GetId() or whatever, but it is useless. Maybe a better way to mask that name will be to set the getID() method as private. In that way it will only be used for the trace. Let me know